src/proc/interface.rs

```diff
@@ -145,16 +150,44 @@ impl<'a> Interface<'a> {
                             crate::Expression::AccessIndex { base, .. } => {
                                 left = base;
                             }
-                            crate::Expression::GlobalVariable(handle) => {
-                                self.uses[handle.index()] |= crate::GlobalUse::STORE;
-                                break;
-                            }
                             _ => break,
                         }
                     }
                     self.add_inputs(value);
                 }
             }
+            self.visitor.sta_visitor(statement, &self.expressions)
```

Comment:
similarly, I think it would make sense to visit the statements first, and then expressions